### PR TITLE
chore: add mergify config for automatic PR management

### PR DESCRIPTION
**Pull request by ryzzaki**
## Description
This PR will introduce Mergify into the repository along with the configuration file. The aim is to automate the PR request to be more elegant when it comes to:

- Merging approved requests
- Deleting merged head branches
- Deleting closed head branches

It should keep our Open Source environment a bit more cleaner & streamlined.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added new practice to practice list in README.md.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - status-success=continuous-integration/travis-ci/push
+      - base=master
+    actions:
+      merge:
+        method: merge
+  - name: delete head branch after merge
+    conditions:
+      - merged
+    actions:
+      delete_head_branch: {}
```
Comment (prokopsimek) [inline .mergify.yml:15]: This part has Github itself in the Settings and it is turned on already. :)

Comment (ryzzaki) [reply .mergify.yml:15]: Oh, apologies. I thought it was being done manually. I will delete this rule. Should I keep the one where head branch is deleted when it is `closed`?

Comment (prokopsimek) [reply .mergify.yml:15]: ✅

**Pull request merged**